auth_proxy/services/oauth.py

```diff
@@ -168,24 +185,15 @@ def cb_tokensetter(self, token, request, *args, **kwargs):
         """
         assert request.user is not None
 
-        try:
-            today = arrow.now().datetime
-            old = self.db.session.query(Token).\
-                filter_by(client_id=request.client.client_id).\
-                filter(Token.approval_expires >= today).\
-                one()
-            old.update_token(**token)
-        except NoResultFound:
-            old = Token(
-                token_type=token['token_type'],
-                _scopes=token['scope'],
-                client_id=request.client.client_id,
-                user=request.user,
-                approval_expires=arrow.now().shift(years=1).datetime,
-            )
-            old.update_token(**token)
-            self.db.session.add(old)
+        today = arrow.now().datetime
+        old = self.db.session.query(Token).\
```

Comment:
So now I'm creating an initial token in `create_authorization`. This happens when the POST to `/oauth/authorize` is successful, because I need data that's only available at that point.

The old `NoResultFound` exception block was run on `/oauth/token`, when a grant token was converted into an access token. Since no access token already existed, it needed to create one. Now there should already be a Token object that it can pick up and update.